StylableSwiftUI/Classes/StylableImage.swift

```diff
@@ -58,352 +56,128 @@ extension Image {
-
-                // Is the bit set?
-                if combinationNumber & (1 << index) > 0 {
-                    // Should this bit affect the value (i.e not a multiple of 2) or affect the variant (i.e. a multiple of 2)
-                    // NB If this index is setting a variant or value which is already a wildcard then we don't need to carry on - it's already been added from when this bit was not set.
-                    if Self.isBitRepresentingValue(index) {
-                        value = self.wildcard
-                    } else {
-                        variant = nil
+                    // Each component can have either the value, the variant, or neither masked out
+                    let valueMask = 1 << (index*2+1)
+
+                    // If we are masking out the value, then we don't care about the variant so just return nil
+                    if self.index & valueMask != 0 {
+                        return nil
                     }
-                }
 
-                Logger.default.log(affectedIndex, index, combinationNumber & (1 << index) > 0 ? "(set)" : "(not set)", component, "->", StylistIdentifier.Component(value: value, variant: variant),
-                                   level: .debug)
+                    let variantMask = 1 << (index*2)
 
-                o[affectedIndex] = StylistIdentifier.Component(value: value, variant: variant)
-            }
+                    // If we are masking out the variant, return a new component
+                    if self.index & variantMask != 0 {
+                        return Component(value: component.value, variant: nil)
+                    }
 
-            combinationNumber += 1
+                    // If we aren't masking anything in this component, just return it as-is
+                    return component
+            }
 
-            let thisVariant = o.map { $0.description }
+            self.index += 1
 
-            variant = thisVariant
+            // If we have a result, is it a duplicate?
+            if let foundResult = result {
+                if deduplicate.contains(foundResult) {
+                    result = nil
+                } else {
+                    deduplicate.insert(foundResult)
+                }
+            }
         }
 
-        return variant
-    }
-
-    // true if a bit set at this index represents a component's value, false if it represents variant
-    private static func isBitRepresentingValue(_ index: Int) -> Bool {
-        return index % 2 != 0
+        return result
     }
 }
 
-extension Sequence {
+public extension StylistIdentifier {
 
-    func leftPad<E>(with prefix: E, count: Int = 1) -> [Element] where Element == [E] {
-        self.map { repeatElement(prefix, count: count) + $0 }
-    }
-}
-
-extension Sequence where Element: Equatable {
+    /// Returns a UIImage instance from the local asset bundle, with the specified user interface traits.
+    func image(separator: String = StylableImage.defaultSeparator,
```

Comment:
Is this a bit of an odd API? i.e use a stylist identifier to get an image?
```
let maybeImage = StylistIdentifier("bla").image
```
🤔 

I'm not sure how easy to use this will be as I feel it should somehow take into account the `currentStylableGroup`? i.e. you have to construct the whole identifier and then get the image. Simplified pseudo-code example:

```swift
var body: some View {
   StylableGroup("test") {
        Image(StylistIdentifier("bla").image)) // won't search for `test_bla`?
    }
}
```

Reply:
(The answer to this might be FU kerr but just thought I'd point out in case you hadn't thought about it)

Reply:
Hmm, OK - is that your use-case above? If so, should we really have an initialiser for Image (i.e. `Image(token:)`) which takes the current group into account?

Or do you need the UIImage for something else? Not sure how to join together UIKit and SwiftUI like that!

Reply:
Or maybe a generic SwiftUI helper which just extracts out the current identifier?

```swift
var body: some View {
   StylableGroup("test") {
        WithIdentifier(token: bla") { identifier in
            Image(identifier.image)) // won't search for `test_bla`?
        }
    }
}
```

That would make it more reusable I guess? And you could then pass out that identifier to anything else (including UIKit?)

Reply:
🤔  Because SwiftUI is still quite new we do some navigation in UIKit. This leads us to customisation of the navigation bar and we need to create UIBarButtonItems (with styled images). Roughly:

```
    public var body: some View {
            someView
                .hiddenNavigationBar(leadingItems: self.buildLeftBarButtons(),
                                      trailingItems: [self.buildShowDetailsButton()],
                                      shouldHideNavigationBar: self.shouldHideNavigationBar)
    }
```
where `hiddenNavigationBar` is a viewModifier that deals with UIKit. In this case I'd like to do something like:

```
    public var body: some View {
            WithIdentifier(tokens: ["foo", "bar"]) { identifier1, id2 in
            someView
                .hiddenNavigationBar(leadingItems: self.buildLeftBarButtons(identifier1),
                                      trailingItems: [self.buildShowDetailsButton(id2)],
                                      shouldHideNavigationBar: self.shouldHideNavigationBar)
}
    }
```
If that looks reasonable?

Reply:
Any objection with replacing 

`WithIdentifier(tokens: ["foo", "bar"]) { identifier1, id2 in` with 

```swift
WithIdentifier(token: "foo") { fooIdentifier in
  WithIdentifier(token: "bar") { barIdentifier in
    someView
                .hiddenNavigationBar(leadingItems: self.buildLeftBarButtons(identifier1),
                                      trailingItems: [self.buildShowDetailsButton(id2)],
                                      shouldHideNavigationBar: self.shouldHideNavigationBar)
  }
}
```

Reply:
I've removed the totally useless extension from this PR to get it merged, and then we can look at this - it's more complicated than I thought!